syntax = "proto3";

package kvstore;

// RPCs for key-value server
service KeyValueStore {
  rpc GetData (GetRequest) returns (GetResponse) {}
  rpc PutData (PutRequest) returns (PutResponse) {}
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 1;
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
  bool success = 1;
}
